### PR TITLE
fix(psalm): psalm fix

### PR DESCRIPTION
**Pull request by skjnldsv**
Signed-off-by: John Molakvoæ <skjnldsv@protonmail.com>

### CODE REVIEW

File: lib/Controller/PreviewController.php [hunk 1]
```diff
@@ -129,7 +129,12 @@ public function index(
 	}
 
 
-	protected function getFileIdForAlbums($fileId, $albums) {
+	/**
+	 * @param (\OCA\Photos\Album\AlbumInfo|mixed)[] $albums
+	 *
+	 * @psalm-param array<\OCA\Photos\Album\AlbumInfo|mixed> $albums
```
Comment (artonge) [inline lib/Controller/PreviewController.php:135]: What is the difference ?

File: lib/Sabre/PhotosHome.php [hunk 2]
```diff
@@ -95,7 +95,9 @@ public function getChild($name) {
 	}
 
 	/**
-	 * @return (AlbumsHome)[]
+	 * @return (AlbumsHome|PlacesHome|SharedAlbumsHome)[]
+	 *
+	 * @psalm-return array{0: AlbumsHome, 1: SharedAlbumsHome, 2: PlacesHome}
```
Comment (artonge) [inline lib/Sabre/PhotosHome.php:100]: Same, what is the difference ? Is `@psalm-return` allowing for a more fine grain type description ? If so, is `@return` still needed ?

File: lib/Sabre/Place/PlaceRoot.php [hunk 3]
```diff
@@ -83,7 +83,9 @@ public function createDirectory($name) {
 	}
 
 	/**
-	 * @return PlacePhoto[]
+	 * @return (PlaceFile|PlacePhoto)[]
```
Comment (artonge) [inline lib/Sabre/Place/PlaceRoot.php:86]: I think it only returns `PlacePhoto`.
Those names are really confusing :see_no_evil:

File: lib/Controller/PreviewController.php (hunk 1 above)
Comment (skjnldsv) [reply lib/Controller/PreviewController.php:135]: This is an automated psalm fix, it knows better than me :see_no_evil:

File: lib/Sabre/Place/PlaceRoot.php (hunk 3 above)
Comment (skjnldsv) [reply lib/Sabre/Place/PlaceRoot.php:86]: If psalm found the two, then something is wrong :wink:

Comment (artonge) [reply lib/Sabre/Place/PlaceRoot.php]: Ah, it is because of the declaration of `$children` !
Fixed :)

**Review by artonge — APPROVED** [review]

**Pull request closed without merging**